### PR TITLE
change kops script to use template

### PR DESCRIPTION
**Pull request by rothgar**
*Description of changes:*

This change updates the development/kops.sh script to use a template instead of requiring manual editing of the cluster file.

This change uses a [kops cluster template](https://github.com/kubernetes/kops/blob/master/docs/operations/cluster_template.md) and generates a values.yaml file to merge config into a `$CLUSTER.yaml` file.

This also will allow us to separate out more variables from the config to allow users to deploy to different regions and change other config.

You should be able to clone this branch and run `./development/kops.sh` and be prompted for all of the necessary information.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: development/eks-d.tpl [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  authorization:
+    rbac: {}
+  channel: stable
+  cloudProvider: aws
+  configBase: {{ .configBase }}
+  containerRuntime: docker
+  etcdClusters:
+  - cpuRequest: 200m
+    etcdMembers:
+    - instanceGroup: master-us-west-2a
```
Comment (micahhausler) [inline development/eks-d.tpl:18]: Some of the 'master' terminology is hard-coded (ex: `masterPublicName` below), but for variable names, lets use 'leader' (unless you think of something better)

And can you template out the region everywhere? 
```suggestion
    - instanceGroup: leader-{{{{.awsRegion}}}}a
```

File: development/eks-d.tpl [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+apiVersion: kops.k8s.io/v1alpha2
+kind: InstanceGroup
+metadata:
+  creationTimestamp: null
+  labels:
+    kops.k8s.io/cluster: {{.clusterName}}
+  name: nodes
+spec:
+  image: 099720109477/ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20201026
+  machineType: m5.xlarge
```
Comment (micahhausler) [inline development/eks-d.tpl:134]:
```suggestion
  machineType: t3.medium
```

File: development/eks-d.tpl [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+apiVersion: kops.k8s.io/v1alpha2
+kind: InstanceGroup
+metadata:
+  creationTimestamp: null
+  labels:
+    kops.k8s.io/cluster: {{.clusterName}}
+  name: master-us-west-2a
+spec:
+  image: 099720109477/ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20201026
+  machineType: m5.large
```
Comment (micahhausler) [inline development/eks-d.tpl:114]: Lets reduce the instance type, no sense in making people pay for something this expensive in a demo (I had originally set it to this for testing)
```suggestion
  machineType: t3.medium
```

**Review by micahhausler — CHANGES_REQUESTED** [review]
Thank you for making my hack more usable!

File: development/eks-d.tpl (hunk 1 above)
Comment (olemarkus) [reply development/eks-d.tpl:18]: `control-plane` is the term we'll migrate towards in kOps, if that helps.

Comment (rothgar) [reply development/eks-d.tpl:18]: I agree there were a lot of variables I wanted to pull out of the template and update names but I also didn't want to change too much in this PR to keep functionality the same and later go through and add some more options.

File: development/eks-d.tpl [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    kops.k8s.io/cluster: {{.clusterName}}
+  name: master-us-west-2a
+spec:
+  image: 099720109477/ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20201026
+  machineType: t3.medium
+  maxSize: 1
+  minSize: 1
+  nodeLabels:
+    kops.k8s.io/instancegroup: master-us-west-2a
+  role: Master
```
Comment (rothgar) [inline development/eks-d.tpl:119]: I'm assuming this is required by kops. Anyone know?

Comment (olemarkus) [reply development/eks-d.tpl:119]: Yes. This one still has to be set to `Master` for control plane nodes.

Comment (micahhausler) [reply development/eks-d.tpl:119]: this is littered [all over the K8s code base](https://github.com/kubernetes/kubernetes/search?q=node-role.kubernetes.io%2Fmaster), and [kops hard codes the values](https://godoc.org/k8s.io/kops/pkg/apis/kops#InstanceGroupRole)

**Review by micahhausler — APPROVED** [review]
/approve
/lgtm

**Pull request merged**